build/build-parse-trees.js

```diff
@@ -58,6 +58,12 @@ names.forEach(function(name) {
   });
   print('   */');
   print('  constructor(%s) {', paramNames.join(', '));
+  if (paramNames[0] === 'location') {
+    print('if (location) {');
```

Comment:
I think I should take this out and re-do this as a pass in test-utils. As the feature tests run they should create an instance of every type and those trees can be checked. Slower only for feature tests that pass.   WDYT?